tests/unit/nupic/algorithms/knn_classifier_test.py

```diff
@@ -29,6 +29,34 @@
 
 class KNNClassifierTest(unittest.TestCase):
 
+ 
+  
+  
+  
+  def testSparsifyVector(self):
+    classifier = KNNClassifier(distanceMethod="norm", distanceNorm=2.0)
+    inputPattern = np.array([0, 1, 3, 7, 11], dtype=np.int32)
+    
+    # Untouched
+    outputPattern = classifier._sparsifyVector(inputPattern, doWinners=True)
+    self.assertTrue(np.array_equal(np.array([0, 1, 3, 7, 11], dtype=np.int32), outputPattern))
+    
+    classifier = KNNClassifier(distanceMethod="norm", distanceNorm=2.0, relativeThreshold=True,
+      sparseThreshold=.2)
+    outputPattern = classifier._sparsifyVector(inputPattern, doWinners=True)
+    self.assertTrue(np.array_equal(np.array([0, 0, 3, 7, 11], dtype=np.int32), outputPattern))
+    
+    classifier = KNNClassifier(distanceMethod="norm", distanceNorm=2.0, relativeThreshold=True,
+      sparseThreshold=.2, numWinners=3)
+    outputPattern = classifier._sparsifyVector(inputPattern, doWinners=True)
+    self.assertTrue(np.array_equal(np.array([0, 0, 3, 7, 11], dtype=np.int32), outputPattern))
+    
+    classifier = KNNClassifier(distanceMethod="norm", distanceNorm=2.0, relativeThreshold=True,
+      sparseThreshold=.2, doBinarization=True)
+    outputPattern = classifier._sparsifyVector(inputPattern, doWinners=True)
+    self.assertTrue(np.array_equal(np.array([0., 0., 1., 1., 1.], dtype=np.int32), outputPattern))
```

Comment:
That seems wrong to me. At the very least it should be well documented. Perhaps @subutai is more familiar with binarization. But I think it is fine to leave this as is for now (outside scope).